modules/claims_api/app/controllers/claims_api/v0/claims_controller.rb

```diff
@@ -17,48 +18,34 @@ def index
       end
 
       def show
-        claim = service.update_from_remote(params[:id])
-        render json: claim, serializer: ClaimsApi::ClaimDetailSerializer
+        fetch_and_render_evss_claim(params[:id])
       end
 
       private
 
-      def service
-        ClaimsApi::UnsynchronizedEVSSClaimService.new(target_veteran)
-      end
-
-      def first_name
-        header(key = 'X-VA-First-Name') ? header(key) : raise_missing_header(key)
-      end
-
-      def last_name
-        header(key = 'X-VA-Last-Name') ? header(key) : raise_missing_header(key)
-      end
-
-      def edipi
-        header(key = 'X-VA-EDIPI') ? header(key) : raise_missing_header(key)
+      def handle_auto_established_claim
+        if ClaimsApi::AutoEstablishedClaim.exists?(params[:id])
+          auto_established_claim = ClaimsApi::AutoEstablishedClaim.find(params[:id])
+          if auto_established_claim.evss_id.nil?
+            render json: auto_established_claim,
+                   serializer: ClaimsApi::AutoEstablishedClaimSerializer
+          else
+            fetch_and_render_evss_claim(auto_established_claim.evss_id)
+          end
+        end
```

Comment:
A verb other than `handle` would be helpful here. Even if I could easily tell what this method _is_ doing, if I'm debugging something a more specific name would be helpful in knowing what the method _should be_ doing.

Separate from the name, I'm a bit confused as to the purpose of this before filter. It's specified globally but it's accessing `params[:id]`. Presumably that would be nil for the `index` action. Should it be specified only for the `show` action? If it is specific to show, then why does the alternative path in the conditional look like it's doing the exact same thing as the `show` action does. Is there some room to de-duplicate code here?

Reply:
100% right on it should only be on the show. The alternate path on the filter / conditional is doing the same thing as show, with a different id. What we're doing is allowing the user to call the end point with either an evss id, OR an `AutoEstablishedClaim` id. If the latter has been assigned the evss id then we query evss for that id. 

I'll take some time thinking about how to make this more clear.

Reply:
I hope this is more clear, I won't resolve til I get some feedback on what you think: 
https://github.com/department-of-veterans-affairs/vets-api/pull/2800/commits/239a508b124645c727248492ab075bcabfbf5cce

Reply:
Much clearer. Thank you!